modules/core/src/lib/attribute.js

```diff
@@ -149,10 +149,12 @@ export default class LayerAttribute extends Attribute {
     if (update) {
       // Custom updater - typically for non-instanced layers
       update.call(context, this, {data, props, numInstances});
-      this.update({
-        value: this.value,
-        constant: this.constant
-      });
+      if (this.gl) {
```

Comment:
* Would it be more readable if we push the `if (gl)` into the `update` method to reduce code length and logic here?
* Any way to make the responsibility of `update` clearer through renaming (`updateWebGLBuffer` etc)?